clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerHeartbeatRequestManager.java

```diff
@@ -97,6 +99,18 @@ public boolean handleSpecificError(final ConsumerGroupHeartbeatResponse response
         boolean errorHandled;
 
         switch (error) {
+            case UNSUPPORTED_VERSION:
+                // Handle consumer-specific unsupported version error
+                String message = CONSUMER_PROTOCOL_NOT_SUPPORTED_MSG;
+                if (errorMessage.contains(REGEX_RESOLUTION_NOT_SUPPORTED_MSG)) {
+                    // If the error is about regex subscription, use the original error message
+                    message = errorMessage;
+                }
```

Comment:
I would expect we don't need this here. This regex error is generated on the client when building the request (it generates a response with exception) so the handling lands on the `onFailure` path (not on the onResponse path that uses this `handleSpeficifError`). 

I would expect we just need to have here the same handling we had  for the UnsupportedVersion in the parent class before this PR:
https://github.com/apache/kafka/blob/ee4264439ddda7bdebcaa845752b824abba14161/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java#L411-L412

That being said, we do need to ensure that we handle responses for each consumer, to cover this bit that is now on the abstract mgr (so it could wrongly apply a consumer msg to the share consumer):

https://github.com/apache/kafka/blob/ee4264439ddda7bdebcaa845752b824abba14161/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java#L322-L325

Both consumers need to ensure they propagate their specific msgs when that exception is in a response.

Reply:
So I guess we should add a handle `onSpeficFailure` in each consumer just like `handleSpecificError` did?